frontend/src/metabase-lib/lib/queries/utils/get-column-key.ts

```diff
@@ -0,0 +1,50 @@
+import { DatasetColumn } from "metabase-types/api";
+import {
+  createFieldReference,
+  getBaseDimensionReference,
+  getDimensionReferenceWithoutOptions,
+  getNormalizedDimensionReference,
+  hasStringFieldName,
+  isAggregationReference,
+  isExpressionReference,
+  isFieldReference,
+  isValidDimensionReference,
+} from "metabase-lib/lib/references";
+
+export const getColumnKey = (
```

Comment:
This is meant to get the column key for the `visualization_settings.columns`, right? I don't get it how this function could work without handling duplicate column names (which would require the entire query). Is this done somewhere else?

Reply:
Currently this logic is in `StructuredQuery` https://github.com/metabase/metabase/blob/25c3050f30cca0309a011e39d15d4350889f05cf/frontend/src/metabase-lib/lib/queries/StructuredQuery.ts#L1405

Reply:
It works with columns coming from the BE which have valid column names:
<img width="759" alt="Screen Shot 2022-10-17 at 1 53 44 PM" src="https://user-images.githubusercontent.com/14301985/196147874-e74e4a79-7721-4341-91e1-4564fceb852f.png">